### PR TITLE
Misc Aesthetic & UX Improvements

### PR DESCRIPTION
**Pull request by alexpaxton**
- [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/Mergable
  - [x] Tests pass

Connect #1566 
Connect #1528 
Connect #1595 

### The Problem
- See above issues
- The theme had its own autoprefixer before being added to the project
- Kapacitor flow slightly confusing
  - Unclear if deleting influxdb source or kapacitor
  - Delete button too close to connect
  - Clarity over which source is connected could be improved
- If a Kapacitor config has a really long name it overlaps with the icons:
  kap-bug-2
- If a Kapacitor shares a name with another Kapacitor (scoped to the same source) the dropdown menu shows all as active
  kap-bug-1

### The Solution
- Remove autoprefixer from theme repo, so that when it comes into Chronograf it is unprefixed, this way there is only one level of autoprefixer and Chronograf's implementation is preferred over the one in the theme
  - Also chopped out more excess styles from the theme
  - `218kb` --> `158kb`
- Remove `width: 100%; display: inline-block;` on type elements, was causing #1566 
- Re-arrange sources table
  - More obvious which source is connected
  - Moved connect buttons to left, feels better for inexplicable reasons
  - Inverted color pattern for `Add Config` buttons vs `Active Kapacitor` dropdowns
  - Inverted color pattern for `Connect` vs `Connected`
  - Visual separator for Kapacitor Column
  - Added `QuestionMarkTooltip` to clarify what the dropdown does
- Single Stat + Line Graph Polish
  - Looks much more like a sparkline (no axes or grid)
  - Line color is always a darker purple
  - Small overlay behind the single stat text for visibility
  - Text shrinks like a regular single stat when the cell is small
- Kapacitor Config UX polish
  - Name is a maximum of 33 chars, which prevents it from overlapping with the edit/delete icons
  - Prevent users from giving a config a name that is already in use

### Screenshots
screen shot 2017-06-13 at 10 07 41 am

screen shot 2017-06-13 at 2 35 28 pm
Shout out to @lukevmorris for the blur idea

screen shot 2017-06-13 at 5 13 38 pm

### CODE REVIEW

File: ui/src/kapacitor/containers/KapacitorPage.js [hunk 1]
```diff
@@ -71,6 +71,23 @@ class KapacitorPage extends Component {
...
+      })
+    }
+    if (doesKapacitorNameAlreadyExist === true) {
+      addFlashMessage({
+        type: 'error',
+        text: `There is already a Kapacitor configuration named "${kapacitor.name}"`,
+      })
+      return
+    }
+
```
Comment (lukevmorris) [inline ui/src/kapacitor/containers/KapacitorPage.js]: Whenever I see a `initialize var`, `map over array to set var`, `switch on final var value`, I always try to rewrite it using a smarter enumerator. In this case, I think we can use `Array.prototype.some`:

```js
let nameExists
if (source && source.kapacitors) {
  nameExists = source.kapacitors.some(k => k.name === kapacitor.name)
}
if (nameExists) {
  ...
```

Comment (lukevmorris) [inline ui/src/kapacitor/containers/KapacitorPage.js]: Another option is to extract the names from each kapacitor object, then ask the result if it includes a known value:
```js
let kapNames = []
if (source && source.kapacitors) {
  kapNames = source.kapacitors.map(k => k.name)
}
if (kapNames.includes(kapacitor.name)) {
```

Comment (lukevmorris) [inline ui/src/kapacitor/containers/KapacitorPage.js]: Also, I looked to see what the source empty state looks like. It appears as though we can assume that a `source` will exist on this page, along with an empty `kapacitors` array if there are no kapacitors. So you can probably remove that check.

Though we should now make explicit in the `propTypes` that we require a kapacitor array. Make the source required, and add an array of kapacitors:
```js
  source: shape({
    id: string.isRequired,
    url: string.isRequired,
    kapacitors: array.isRequired,
  }).isRequired,
```

File: ui/src/sources/components/InfluxTable.js [hunk 2]
```diff
@@ -90,27 +96,59 @@ const InfluxTable = ({
...
               <tr>
-                <th>Name</th>
-                <th>Host</th>
-                <th>Kapacitor</th>
+                <th className="source-table--connect-col" />
+                <th>Source Name & Host</th>
                 <th className="text-right" />
+                <th>
+                  Active Kapacitor{' '}
+                  <QuestionMarkTooltip
```
Comment (lukevmorris) [inline ui/src/sources/components/InfluxTable.js]: Sweeeeeeeeet

File: ui/src/style/components/dygraphs.scss [hunk 3]
```diff
@@ -153,13 +153,37 @@
...
+  top: 50%;
+  left: 50%;
+  width: 110%;
+  height: 0;
+  transform: translate(-50%,-50%);
+  box-shadow: fade-out($g2-kevlar, 0.3) 0 0 50px 30px;
+  z-index: -1;
+}
+.single-stat--small .single-stat--shadow:after {
+  box-shadow: fade-out($g2-kevlar, 0.3) 0 0 30px 10px;
```
Comment (lukevmorris) [inline ui/src/style/components/dygraphs.scss]: Killer

File: ui/src/kapacitor/containers/KapacitorPage.js (hunk 1 above)
Comment (lukevmorris) [inline ui/src/kapacitor/containers/KapacitorPage.js]: Making this assertion lets us remove the `source && source.kapacitors` check, and this codeblock becomes super simple

Comment (alexpaxton) [inline ui/src/kapacitor/containers/KapacitorPage.js]: Made `kapacitors` a required array (as above), and then used `.includes` to detect duplicates. Much cleaner, thanks!

**Comment by jaredscheib** [comment]
Terrific work @alexpaxton. Exciting to see your iterations in JS, in addition to the evolving design and style work!

**Pull request merged**